server/controller/usersControlloer.js

```diff
@@ -0,0 +1,181 @@
+const dotenv = require("dotenv");
+dotenv.config();
+const conn = require("../mariadb");
+const { StatusCodes } = require("http-status-codes");
+const jwt = require("jsonwebtoken");
+const crypto = require("crypto");
+const ensureAuthorization = require("../auth");
+
+const join = (req, res) => {
+  const { email, password, location, nickname } = req.body;
+
+  const now = new Date();
+  const formattedTime = new Date(now);
+
+  let sql = `INSERT INTO users (email, password, salt, location, nickname, created_time) VALUES (?, ?, ?, ?, ?, ?)`;
+
+  const salt = crypto.randomBytes(10).toString("base64");
+  const hashPassword = crypto
+    .pbkdf2Sync(password, salt, 10000, 10, "sha512")
+    .toString("base64");
+
+  let values = [email, hashPassword, salt, location, nickname, formattedTime];
+  conn.query(sql, values, (err, results) => {
+    if (err) {
+      console.log(err);
+      return res.status(StatusCodes.BAD_REQUEST).end();
+    }
+
+    if (results.affectedRows) {
+      return res.status(StatusCodes.CREATED).json(results);
+    } else {
+      return res.status(StatusCodes.BAD_REQUEST).end();
+    }
+  });
+};
+
+const login = (req, res) => {
+  const { email, password } = req.body;
+
+  let sql = `SELECT * FROM users WHERE email = ?`;
+  conn.query(sql, email, (err, results) => {
+    if (err) {
+      console.log(err);
+      return res.status(StatusCodes.BAD_REQUEST).end();
+    }
+
+    const loginUser = results[0];
+
+    // salt값 꺼내서 날 것으로 들어온 비밀번호 암호화
+    const hashPassword = crypto
+      .pbkdf2Sync(password, loginUser.salt, 10000, 10, "sha512")
+      .toString("base64");
+
+    // 위에서 암호화된 비밀번호 db 비밀번호랑 비교
+    if (loginUser && loginUser.password == hashPassword) {
+      const token = jwt.sign(
+        {
+          id: loginUser.id,
+          email: loginUser.email,
+        },
+        process.env.PRIVATE_KEY,
+        {
+          expiresIn: "1h",
```

Comment:
이런 값은 상수로 다뤄주면 좋습니다~
```suggestion
          expiresIn: TOKEN_EXPIRED_TIME,
```